pandas/io/excel/_base.py

```diff
@@ -380,7 +380,7 @@ def get_sheet_by_index(self, index):
         pass
 
     @abc.abstractmethod
-    def get_sheet_data(self, sheet, convert_float):
+    def get_sheet_data(self, sheet, convert_float, header, skiprows, nrows):
```

Comment:
```suggestion
    def get_sheet_data(self, sheet, convert_float, header: int, skiprows: int, nrows: int):
```

We try to annotate all new arguments, so would be good to do here (I'm assuming they are all int)

Reply:
Both header and skiprows can be a list, but this optimization works only if all three variables are ints. nrows is always int. Should I annotate only nrows?

Reply:
If you can annotate everything new that you’ve added with the appropriate types that would be helpful. Sounds like the others should be `Union[int, Sequence[int]]` then

Reply:
Done